### DISCUSSION TITLE
Can I use Sveltia-cms work with a Github-pages site without hosting on netlify?

### DISCUSSION THREAD

**Discussion by RDjarbeng** [sveltia/sveltia-cms#218] [Q&A] [answered]
I want to know if it's possible to use sveltia without hosting my site on netlify?
 
 This is a [question](https://github.com/decaporg/decap-website/issues/79) I asked on the Decap-CMS issue page that I hope can be solved here.

The short form of the question is that,  I am hosting my site on GitHub-pages, but I have had to deploy to netlify just to be able to use the CMS. 
I understood from DecapCMS that this is because of the netlify identity widget, but since Sveltia  CMS doesn't use netlify identity I was hoping this could be done.
State of my current setup 

my-site.github-pages - here I view the site
my-site.github-pages/admin  - this doesn't work currently, added a screenshot

![image](https://github.com/user-attachments/assets/f3ace622-491b-4e35-88a2-990e824e0d01)

my-site.netlify.app/admin - This is the netlify site where I access the CMS

The problem with this is that it gives me two platforms to manage (github-pages and netlify) I was hoping I would be able to just stay with one.

Here is my site repo in case it helps:

https://github.com/RDjarbeng/RDjarbeng

**Comment by kyoshino** [comment, marked as answer]
Both Netlify/Decap CMS and Sveltia CMS can be hosted anywhere. But yeah, the documentation is not clear as discussed in your Decap ticket. Sveltia CMS will offer better documents to avoid confusion once it hits the GA. (Especially if you don’t know Decap CMS was Netlify CMS, things might be super confusing)

Basically, you have to move away from the [Git Gateway backend](https://decapcms.org/docs/git-gateway-backend/) + Netlify Identity combo to the [GitHub backend](https://decapcms.org/docs/github-backend/) + an [external OAuth client](https://decapcms.org/docs/external-oauth-clients/).

We provide our own client called [Sveltia CMS Authenticator](https://github.com/sveltia/sveltia-cms-auth): follow the instructions and you should be able to sign in with GitHub. 🙂

**Comment by kyoshino** [comment]
We should better explain these different layers 😉 

- Repository: GitHub or GitLab
- Authentication: Git Gateway + Netlify Identity, Sveltia CMS Authenticator or other OAuth client
- Building: GitHub Actions, GitLab CI/CD, Cloudflare Pages, Netlify, Vercel, etc.
- Hosting: GitHub Pages, GitLab Pages, Cloudflare Pages, Netlify, Vercel, etc.

**Comment by RDjarbeng** [comment]
Thanks will try this and see if it works. Hope to report back with good news

**Comment by RDjarbeng** [comment]
Okay the deploy for the [sveltia-cms-auth](https://deploy.workers.cloudflare.com/?url=https://github.com/sveltia/sveltia-cms-auth) took a while because there seems to be missing steps on the Cloudflare page to get the account id and API key.

Also the instructions to get the account id are no longer valid which is in this [issue](https://github.com/cloudflare/cloudflare-docs/issues/12493#issuecomment-2399552721) here. What worked for me was to copy the id from the link on the Cloudflare dashboard instead.

Apparently you need to create a new API key with permissions to edit the worker, not the global API key one which will not work. One hurdle down.🚶🏽‍♂️

The small hiccup when trying to deploy the worker is more of a problem with the documentation for Cloudflare on how to get the account id and use the right API key, which is Not really a problem with [Sveltia CMS Authenticator](https://github.com/sveltia/sveltia-cms-auth?tab=readme-ov-file)


PS: 
May lead to multiple open tabs to complete this step😕
![image](https://github.com/user-attachments/assets/4c66e645-1203-4459-84dc-6793371592c5)

**Reply by kyoshino** [reply]
I wasn’t aware of the documentation issues with Cloudflare Pages, but I understand that setting up the Authenticator is complicated anyway. I’ll figure out other ways to authenticate, like what’s suggested in:

- https://github.com/decaporg/decap-cms/issues/7108

**Comment by RDjarbeng** [comment]
It worked.
![nft-shellz](https://github.com/user-attachments/assets/76697a54-49c5-4d2b-97d4-60cadd0500c3)

Now I am able to access the CMS at mysite/admin on GitHub pages, instead of going to the mysite.netlify.app/admin


Thank you @kyoshino
